internal/pkg/service/buffer/storage/local/reader/volume.go

```diff
@@ -94,18 +92,22 @@ func (v *Volume) Close() error {
 	errs := errors.NewMultiError()
 	v.logger.Info("closing volume")
 
-	// Cancel all operations
+	// Block NewReaderFor method
 	v.cancel()
 
-	// Close all slice writers
-	for _, w := range v.openedReaders() {
-		if err := w.Close(); err != nil {
-			errs.Append(errors.Errorf(`cannot close reader for slice "%s": %w`, w.SliceKey().String(), err))
-		}
+	// Close all slice readers
+	wg := &sync.WaitGroup{}
+	for _, r := range v.openedReaders() {
+		r := r
+		wg.Add(1)
+		go func() {
```

Comment:
Ukoncime `readers` paralelne, bude ich tolko ako slices, teda `100x+`, a sekvencne by to mohlo trvat zbytocne dlho.